app-backup/luckybackup/patches/luckybackup-0.4.9.patchset

```diff
@@ -0,0 +1,84 @@
+From ab3fd8dfc57e6240ffcba7605a6c9602125b00cb Mon Sep 17 00:00:00 2001
+From: mazbrili <mazbrili@yahoo.com>
+Date: Fri, 5 Oct 2018 09:28:19 +0700
+Subject: set current dir to application dir
+
+
+diff --git a/src/main.cpp b/src/main.cpp
+index 935603e..4f66bba 100644
+--- a/src/main.cpp
++++ b/src/main.cpp
+@@ -33,6 +33,9 @@
+ 
+ #include <QApplication>
+ #include <QTranslator>
++#ifdef Q_OS_HAIKU
++#include <QDir>
++#endif
+ 
+ #include "luckybackupwindow.h"
+ #include "global.h"
+@@ -52,6 +55,10 @@ int main(int argc, char *argv[])
+ {
+     QScopedPointer<QCoreApplication> app(createApplication(argc, argv));
+ 
++#ifdef Q_OS_HAIKU
++    QDir::setCurrent(QCoreApplication::applicationDirPath());
++#endif
++
+     //arguments test
+     if (!argumentsTest(argc,argv))
+         return 1;		// error code1: something wrong with the command arguments - app fail to start (or just --help or --version given as args)
+-- 
+2.19.0
+
+
+From 8ea9eea609a179dfe67852cfab86250d10142f83 Mon Sep 17 00:00:00 2001
+From: fbrosson <fbrosson@localhost>
+Date: Wed, 17 Oct 2018 14:23:21 +0000
+Subject: hardcode paths for Haiku
+
+
+diff --git a/src/global.cpp b/src/global.cpp
+index 1efe3ed..3e0cb0f 100644
+--- a/src/global.cpp
++++ b/src/global.cpp
+@@ -52,8 +52,8 @@ QString appVersionString = "0.4.9";
+ double const validProfileVersion = 0.21;
+ double const validScheduleVersion = 0.34;
+ double const validSettingsVersion = 0.3;
+-QString luckyBackupDir = myHome + "/."+appName+"/";
+-QString luckyBackupDefaultDir = myHome + "/."+appName+"/";
++QString luckyBackupDir = myHome + "/config/settings/LuckyBackup/";
++QString luckyBackupDefaultDir = myHome + "/config/settings/LuckyBackup/";
```

Comment:
[`QStandardPaths`](https://doc.qt.io/qt-5/qstandardpaths.html), or more specifically `QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation)` should be used.

The usage is simple, just

    luckyBackupDir = QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation)

simpler than our `find_directory` in fact... And it's even cross-platform.

Reply:
Cool! This can become an excellent GCI task!
(I'll leave this to another mentor :)